docs/src/content/docs/get-started/url-shortner.mdx

```diff
@@ -0,0 +1,306 @@
+Syntax - `Set(key, value, expiration)`
+`key`: Unique identifier (e.g., short URL code)
+`value`: The data to store (serialized JSON)
+`expiration`: Optional; 0 means no expiration
+
+2. `Get` Command: Retrieves the value associated with a key.
+Syntax - `Get(key)`
+`key`: The identifier for the data to retrieve.
+
+## Project Structure
+
+The project is organized into several packages:
+
+- `models`: Defines the URL data structure.
+- `repository`: Handles database interactions with DiceDB.
+- `services`: Implements business logic for URL shortening.
+- `handlers`: Contains HTTP route handlers for the API.
+- `utils`: Provides helper functions for API responses.
+
+## Code overview
+
+1. `models/url.go`: Defines the URL struct, which represents the data to be stored in DiceDB.
+
+    ```go
+    package models
+
+    type URL struct {
+        ID       string `json:"id"`
+        LongURL  string `json:"long_url"`
+        ShortURL string `json:"short_url"`
+    }
+    ```
+
+    - This struct will be serialized to JSON before saving in DiceDB.
+
+2. `repository/url_repository.go`: Handles interactions with DiceDB, using `Set` and `Get` commands to store and retrieve URL data.
+
+    ```go
+    package repository
+
+    import (
+        "context"
+        "encoding/json"
+        "errors"
+        "fmt"
+        "url-shortner-dicedb/models"
+
+        "github.com/dicedb/dicedb-go" // DiceDB Go SDK
+    )
+
+    var db *dicedb.Client
+
+    func init() {
+        db = dicedb.NewClient(&dicedb.Options{
+            Addr: "localhost:7379", 
+        })
+    }
+
+    // SaveURL stores a URL in DiceDB
+    func SaveURL(url models.URL) error {
+        ctx := context.Background()
+
+        // Serialize the URL struct to JSON
+        urlData, err := json.Marshal(url)
+        if err != nil {
+            return fmt.Errorf("failed to serialize URL: %w", err)
+        }
+
+        // Store the serialized URL JSON in DiceDB
+        statusCmd := db.Set(ctx, url.ID, urlData, 0)
+        _, err = statusCmd.Result()
+        if err != nil {
+            return fmt.Errorf("failed to save URL: %w", err)
+        }
+        return nil
+    }
+
+    // FindURLByID retrieves a URL by its ID
+    func FindURLByID(id string) (models.URL, error) {
+        ctx := context.Background()
+
+        // Retrieve the serialized URL data from DiceDB
+        stringCmd := db.Get(ctx, id)
+        urlData, err := stringCmd.Result()
+        if err != nil {
+            if err == dicedb.Nil {
+                return models.URL{}, errors.New("URL not found")
+            }
+            return models.URL{}, fmt.Errorf("failed to retrieve URL: %w", err)
+        }
+
+        // Deserialize the JSON data back into the URL struct
+        var url models.URL
+        err = json.Unmarshal([]byte(urlData), &url)
+        if err != nil {
+            return models.URL{}, fmt.Errorf("failed to decode URL data: %w", err)
+        }
+        return url, nil
+    }
```

Comment:
Based on above comments this can be simplified down to just a couple of lines of code. All it needs to do is lookup keys by short URL.